tasks/migrate-certificates.ts

```diff
@@ -106,104 +114,82 @@ export const GET_MIGRATE_CERTIFICATES_TASK = () =>
-            const tokenId = parseTransactionLogs({
+            txReceipt = await removalContract.provider.getTransactionReceipt(
+              result.transactionHash
+            );
+            tokenId = parseTransactionLogs({
               contractInstance: removalContract,
               txReceipt,
             })
               .filter((log) => log.name === 'Migrate')
               .flatMap((log) => log.args.certificateId.toNumber())[0];
-
             if (txReceipt.status !== 1) {
               // log an error that this transaction hash failed and we are exiting early
-              hre.log(
-                chalk.red(
-                  `❌ Transaction ${pendingTx.hash} failed with failure status ${txReceipt.status} - exiting early`
-                )
+              logger.error(
+                `❌ Transaction ${pendingTx.hash} failed with failure status ${txReceipt.status} - exiting early`
               );
               return;
             }
-            PROGRESS_BAR.increment();
-            outputData.push({
-              ...certificate,
-              txReceipt,
-              tokenId,
-            });
-          } catch (error) {
-            PROGRESS_BAR.stop();
-            hre.log(
-              chalk.red(
-                `❌ Error minting certificate ${certificate.id} (number ${certificateIndex}/${jsonData.length}) - exiting early`
-              )
-            );
-            hre.log(error);
-            outputData.push({
-              ...certificate,
-              error,
-            });
-            writeJsonSync(outputFile, outputData);
-            return;
-          }
-        } else {
-          // dry run
-          try {
-            await removalContract.callStatic.migrate(
-              certificate.ids,
-              amounts,
-              signerAddress, // TODO make sure this is the Nori admin address that can transfer certificates
-              ethers.utils
-                .parseUnits(certificate.data.numberOfNrts.toString())
-                .toString()
-            );
-            hre.log(
-              `(${certificateIndex}/${jsonData.length}) projectId: ${certificate.id} 🎉  Dry run was successful!`
-            );
-          } catch {
-            hre.log(
-              `(${certificateIndex}/${jsonData.length}) projectId: ${certificate.id} 💀 Dry run was unsuccessful!`
-            );
           }
+          PROGRESS_BAR.increment();
+          outputData.push({
+            ...certificate,
+            txReceipt,
+            tokenId,
+          });
+        } catch (error) {
+          PROGRESS_BAR.stop();
+          logger.error(
+            `❌ Error minting certificate ${
+              JSON.parse(certificate.key).id
+            } (number ${certificateIndex}/${jsonData.length}) - exiting early`
+          );
+          hre.log(error);
+          outputData.push({
+            ...certificate,
+            error,
+          });
+          writeJsonSync(outputFileName, outputData);
```

Comment:
@amiecorso do you recall if/why this second write is needed since we're not throwing? Could the error just be initialized pre-loop and then conditionally added to the final write call?